lldb/docs/man/lldb-server.rst

```diff
@@ -147,15 +147,8 @@ GDB-SERVER CONNECTIONS
 
 .. option:: --gdbserver-port <port>
 
- Define a port to be used for gdb-server connections. Can be specified multiple
- times to allow multiple ports. Has no effect if --min-gdbserver-port
- and --max-gdbserver-port are specified.
-
-.. option:: --min-gdbserver-port <port>
-.. option:: --max-gdbserver-port <port>
-
- Specify the range of ports that can be used for gdb-server connections. Both
- options need to be specified simultaneously. Overrides --gdbserver-port.
+ Define a port to be used for gdb-server connections. This port is used for
+ multiple connections.
```

Comment:
I don't know much about how ports work, can you explain briefly how that single port knows to route traffic from each client lldb to the right lldb-server gdbserver?

It appears to "just work" but I'm curious.

Reply:
Before: 
- `lldb-server platform`
  - got the query `qLaunchGDBServer`
  - request a gdb port from the map (it may be 0)
  - launch `lldb-server gdbserver host:port`
  - if the port was 0, read the actual port back via pipe
  - answer OK and report the actual port and pid
- lldb client connects to gdbserver using host:port (if handshake failed, lldb client kills gdbserver by pid via platform protocol)

After:
- `lldb-server platform`
  - listen to gdbserver-port
  - got the query `qLaunchGDBServer`
  - answer OK and report always known gdbserver-port and ibvalid pid
- lldb client connects to gdbserver using host:port (do nothing if handshake failed)
- `lldb-server platform` got connection and launch `lldb-server gdbserver --fd socket`
  - `lldb-server platform` will transfer the socket info to gdbserver via pipe on Windows (or kill gdbserver if it failed)

Reply:
Thanks, I understood that bit.

What I was missing was basic socket theory, that on the remote side we can look at the IP address of the sender so IP A to port N can be differentiated from IP B sending to port N. It's been a very long time since I had to think about this, but in hindsight, of course lots of things multiplex this way.

Nothing wrong with the code, if anything it was a bit too magical :) And a good improvement for the way I use lldb in my work.